<launch>
  <!-- image sync -->
  <node pkg="perception_interface" type="image_synchronize.py" name="depth_sync" output="screen"/>

  <!-- image republish -->  
  <node name="color_republish" type="republish" pkg="image_transport" output="screen" 
        args="compressed in:=/camera/color/image_rect_color raw out:=/camera/color/image_rect_color/republish" />

  <!-- Nodelet manager for this pipeline -->
  <node pkg="nodelet" type="nodelet" args="manager"
        name="depth_manager" output="screen"/>


  <!-- Convert it into a XYZRGB point cloud -->
  <node pkg="nodelet" type="nodelet" name="cloudify"
        args="load depth_image_proc/point_cloud_xyzrgb depth_manager --no-bond">
    <remap from="depth_registered/points"     to="camera/test/points"/>
    <remap from="rgb/camera_info"             to="/camera/test/rgb_info"/>
    <remap from="rgb/image_rect_color"        to="/camera/test/rgb"/>
    <remap from="depth_registered/image_rect" to="/camera/test/depth"/>
  </node>

</launch>